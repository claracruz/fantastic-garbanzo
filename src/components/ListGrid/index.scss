.list-grid {
    list-style-type: none;
    padding: 0;
}
