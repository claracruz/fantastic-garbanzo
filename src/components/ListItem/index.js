import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';
import {faStarHalfAlt} from '@fortawesome/free-solid-svg-icons'
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

const months = [
	'Jan',
	'Feb',
	'Mar',
	'Apr',
	'May',
	'Jun',
	'Jul',
	'Aug',
	'Sep',
	'Oct',
	'Nov',
	"'Dec"
];


export const ListItem = (props) => {
	const {altmetric_id, details_url, images, published_on, score, title, url} = props.item;
	const datePublished = (published_on) ? new Date(published_on * 1000) : null;

	return (
		<li key={altmetric_id} className="list-item">
			<img src={images.small} alt={title}/>
			<span className="date-published">
				<span className="day">{datePublished && datePublished.getDay()}</span>
		        <span className="month">{months[datePublished && datePublished.getMonth()]}</span>
		        <span className="year">{datePublished && datePublished.getFullYear()}</span>
			</span>
			<div className="content">
				<h3>{title}</h3>

				<div className="details">
					<span className="score">
						<FontAwesomeIcon icon={faStarHalfAlt} />
						<span>{score}</span>
					</span>
					<a href={details_url}
					   className="source"
					   target="_blank"
					   rel="noopener noreferrer" >View metrics data...</a>
					<a href={url}
					   className="source"
					   target="_blank"
					   rel="noopener noreferrer">Go to source...</a>
				</div>
			</div>
		</li>
	);
};


ListItem.propTypes = {
	item: PropTypes.shape({
		altmetric_id: PropTypes.number,
		details_url: PropTypes.string,
		title: PropTypes.string,
		url: PropTypes.string,
		published_on: PropTypes.number,
		score: PropTypes.number,
		images: PropTypes.shape({
			small: PropTypes.string,
			medium: PropTypes.string,
			large: PropTypes.string
		})
	})
};
