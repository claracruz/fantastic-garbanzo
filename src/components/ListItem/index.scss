.list-item {

  background: #ffffff;
  height: auto;
  overflow: hidden;
  width: auto;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.3);
  margin: 15px 0;
  padding: 20px 20px 20px 75px;
  position: relative;
  display: flex;
  text-align: left;

  img {
    flex-shrink: 0;
    width: 64px;
    height: 64px;
  }

  .date-published {
    position: absolute;
    top: 0;
    left: 0;
    background-color: #77d7b9;
    color: #fff;
    padding: 0.8em;

    span {
      display: block;
      text-align: center;
      font-size: 0.8125rem;

      &.day {
        font-weight: 700;
        font-size: 24px;
        text-shadow: 2px 3px 2px rgba(0, 0, 0, 0.18);
      }

      &.month {
        text-transform: uppercase;
      }
    }
  }

  .content {
    margin: 0 25px;

    h3 {
      font-size: 1em;
      width: 75%;
      color: #444;
    }

    a {
      text-decoration: none;

      &:hover {
        text-decoration: underline;
      }
    }

    .source {
      color: #0198A6;
      font-size: 0.8125rem;
      font-weight: bold;
      margin-left: 20px;
    }

    .details {
      margin-top: 15px;
      font-size: 0.8125rem;
      font-weight: bold;
      color: #0198A6;

      .score {

        span, a {
          position: relative;
          display: inline-block;
          left: 3px;
          top: 1px;
        }
      }
    }

  }
}
