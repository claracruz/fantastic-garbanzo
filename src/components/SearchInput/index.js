import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import './index.scss';

export class SearchInput extends PureComponent {

	constructor(props) {
		super(props);
		this.state = {
			searchTerm: ''
		}
	}

	onSubmit = () => {
		this.props.onSearch(this.state.searchTerm);
	};

	handleChange = (e) => {
		this.setState({
			searchTerm: e.currentTarget.value
		});
	};

	handleKeyDown = (e) => {
		if (e.key === 'Enter') {
			this.onSubmit();
		}
	};

	render() {
		return (
			<div className="wrap">
				<h2><label htmlFor="search">Search Papers</label></h2>
				<div className="search">
					<input
						id="search"
						className="search-input"
						value={this.state.searchTerm}
						onKeyPress={this.handleChange}
						onKeyDown={this.handleKeyDown}
						onChange={this.handleChange}
						placeholder="Enter a search term"
						type="text" />
					<button className="searchButton" onClick={this.onSubmit}><FontAwesomeIcon icon={faSearch} /></button>
				</div>
			</div>
		);
	}
}

SearchInput.propTypes = {
	onSearch: PropTypes.func.isRequired
};
