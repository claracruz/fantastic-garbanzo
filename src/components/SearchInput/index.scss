
.wrap {
    width: 100%;
    display: flex;
    justify-content: flex-start;
    padding: 20px;
    height: 65px;

    h2 {
        margin-right: 40px;
        position: relative;
        top: 3px;
    }

    .search {
        max-width: 320px;
        width: 100%;
        position: relative;
        margin-right: 35px;
    }

    .search-input {;
        width: 100%;
        border: 3px solid #0198A6;
        padding: 5px;
        height: 36px;
        border-radius: 5px;
        outline: none;
        font-size: 0.8125rem;
        background-color: white;
    }

    .searchButton {
        position: absolute;
        right: -32px;
        width: 40px;
        height: 36px;
        border: 1px solid #0198A6;
        background: #0198A6;
        text-align: center;
        color: #fff;
        border-top-right-radius: 5px;
        border-bottom-right-radius: 5px;
        cursor: pointer;
        font-size: 1rem;
    }
}


