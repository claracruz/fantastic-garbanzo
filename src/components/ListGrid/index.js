import React  from 'react';
import PropTypes from 'prop-types';
import { ListItem } from '../ListItem';
import './index.scss';


function displayData (props) {
	const { items } = props;
	return items.map((item) => {
		return (
			<ListItem key={item.altmetric_id} item={item}/>
		);
	})
}

export const ListGrid = (props) => {
	const { className, items } = props;
	const hasData = items.length > 0;

	return (
		<>
			{
				hasData &&
				<ul className={className}>{displayData(props)}</ul>
			}
		</>
	);
};

ListGrid.propTypes = {
	items: PropTypes.arrayOf(
		PropTypes.shape({
			altmetric_id: PropTypes.number,
			title: PropTypes.string,
			url: PropTypes.string,
			details_url: PropTypes.string,
			published_on: PropTypes.number,
			score: PropTypes.number,
			images: PropTypes.shape({
				small: PropTypes.string,
				medium: PropTypes.string,
				large: PropTypes.string
			}),
		})
	)
};

ListGrid.defaultProps = {
	items: []
};
