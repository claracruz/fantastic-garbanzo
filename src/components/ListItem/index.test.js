import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { ListItem } from '../../components/ListItem';

describe('<ListItem />', () => {

	let props;

	beforeEach(() => {
		props = {
			item: {
				altmetric_id: 112,
				details_url: 'http://url/detail',
				title: 'title',
				url: 'http://url',
				published_on: 430,
				score: 20,
				images: {
					small: '/img/path/small',
					medium: '/img/path/medium',
					large: '/img/path/large',
				}
			}
		};
	});

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<ListItem {...props} />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders correctly', () => {
		const wrapper = shallow(<ListItem {...props} />);
		const instance = wrapper.instance();
		expect(wrapper).toMatchSnapshot();
		expect(instance).toBe(null);

	});
});
