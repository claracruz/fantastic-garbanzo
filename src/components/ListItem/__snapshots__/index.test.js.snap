// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    item={
      Object {
        "altmetric_id": 112,
        "details_url": "http://url/detail",
        "images": Object {
          "large": "/img/path/large",
          "medium": "/img/path/medium",
          "small": "/img/path/small",
        },
        "published_on": 430,
        "score": 20,
        "title": "title",
        "url": "http://url",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "112",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="title"
          src="/img/path/small"
        />,
        <span
          className="date-published"
        >
          <span
            className="day"
          >
            4
          </span>
          <span
            className="month"
          >
            Jan
          </span>
          <span
            className="year"
          >
            1970
          </span>
        </span>,
        <div
          className="content"
        >
          <h3>
            title
          </h3>
          <div
            className="details"
          >
            <span
              className="score"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      536,
                      512,
                      Array [],
                      "f5c0",
                      "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                    ],
                    "iconName": "star-half-alt",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              <span>
                20
              </span>
            </span>
            <a
              className="source"
              href="http://url/detail"
              rel="noopener noreferrer"
              target="_blank"
            >
              View metrics data...
            </a>
            <a
              className="source"
              href="http://url"
              rel="noopener noreferrer"
              target="_blank"
            >
              Go to source...
            </a>
          </div>
        </div>,
      ],
      "className": "list-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "title",
          "src": "/img/path/small",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="day"
            >
              4
            </span>,
            <span
              className="month"
            >
              Jan
            </span>,
            <span
              className="year"
            >
              1970
            </span>,
          ],
          "className": "date-published",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "className": "day",
            },
            "ref": null,
            "rendered": 4,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Jan",
              "className": "month",
            },
            "ref": null,
            "rendered": "Jan",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1970,
              "className": "year",
            },
            "ref": null,
            "rendered": 1970,
            "type": "span",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              title
            </h3>,
            <div
              className="details"
            >
              <span
                className="score"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        536,
                        512,
                        Array [],
                        "f5c0",
                        "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                      ],
                      "iconName": "star-half-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span>
                  20
                </span>
              </span>
              <a
                className="source"
                href="http://url/detail"
                rel="noopener noreferrer"
                target="_blank"
              >
                View metrics data...
              </a>
              <a
                className="source"
                href="http://url"
                rel="noopener noreferrer"
                target="_blank"
              >
                Go to source...
              </a>
            </div>,
          ],
          "className": "content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "title",
            },
            "ref": null,
            "rendered": "title",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="score"
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          536,
                          512,
                          Array [],
                          "f5c0",
                          "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                        ],
                        "iconName": "star-half-alt",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                  <span>
                    20
                  </span>
                </span>,
                <a
                  className="source"
                  href="http://url/detail"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View metrics data...
                </a>,
                <a
                  className="source"
                  href="http://url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Go to source...
                </a>,
              ],
              "className": "details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <FontAwesomeIcon
                      border={false}
                      className=""
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            536,
                            512,
                            Array [],
                            "f5c0",
                            "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                          ],
                          "iconName": "star-half-alt",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    />,
                    <span>
                      20
                    </span>,
                  ],
                  "className": "score",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "border": false,
                      "className": "",
                      "fixedWidth": false,
                      "flip": null,
                      "icon": Object {
                        "icon": Array [
                          536,
                          512,
                          Array [],
                          "f5c0",
                          "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                        ],
                        "iconName": "star-half-alt",
                        "prefix": "fas",
                      },
                      "inverse": false,
                      "listItem": false,
                      "mask": null,
                      "pull": null,
                      "pulse": false,
                      "rotation": null,
                      "size": null,
                      "spin": false,
                      "symbol": false,
                      "title": "",
                      "transform": null,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 20,
                    },
                    "ref": null,
                    "rendered": 20,
                    "type": "span",
                  },
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "View metrics data...",
                  "className": "source",
                  "href": "http://url/detail",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "View metrics data...",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Go to source...",
                  "className": "source",
                  "href": "http://url",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Go to source...",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "112",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="title"
            src="/img/path/small"
          />,
          <span
            className="date-published"
          >
            <span
              className="day"
            >
              4
            </span>
            <span
              className="month"
            >
              Jan
            </span>
            <span
              className="year"
            >
              1970
            </span>
          </span>,
          <div
            className="content"
          >
            <h3>
              title
            </h3>
            <div
              className="details"
            >
              <span
                className="score"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        536,
                        512,
                        Array [],
                        "f5c0",
                        "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                      ],
                      "iconName": "star-half-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span>
                  20
                </span>
              </span>
              <a
                className="source"
                href="http://url/detail"
                rel="noopener noreferrer"
                target="_blank"
              >
                View metrics data...
              </a>
              <a
                className="source"
                href="http://url"
                rel="noopener noreferrer"
                target="_blank"
              >
                Go to source...
              </a>
            </div>
          </div>,
        ],
        "className": "list-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "title",
            "src": "/img/path/small",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="day"
              >
                4
              </span>,
              <span
                className="month"
              >
                Jan
              </span>,
              <span
                className="year"
              >
                1970
              </span>,
            ],
            "className": "date-published",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 4,
                "className": "day",
              },
              "ref": null,
              "rendered": 4,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Jan",
                "className": "month",
              },
              "ref": null,
              "rendered": "Jan",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1970,
                "className": "year",
              },
              "ref": null,
              "rendered": 1970,
              "type": "span",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                title
              </h3>,
              <div
                className="details"
              >
                <span
                  className="score"
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          536,
                          512,
                          Array [],
                          "f5c0",
                          "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                        ],
                        "iconName": "star-half-alt",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />
                  <span>
                    20
                  </span>
                </span>
                <a
                  className="source"
                  href="http://url/detail"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  View metrics data...
                </a>
                <a
                  className="source"
                  href="http://url"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Go to source...
                </a>
              </div>,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "title",
              },
              "ref": null,
              "rendered": "title",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="score"
                  >
                    <FontAwesomeIcon
                      border={false}
                      className=""
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            536,
                            512,
                            Array [],
                            "f5c0",
                            "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                          ],
                          "iconName": "star-half-alt",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size={null}
                      spin={false}
                      symbol={false}
                      title=""
                      transform={null}
                    />
                    <span>
                      20
                    </span>
                  </span>,
                  <a
                    className="source"
                    href="http://url/detail"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    View metrics data...
                  </a>,
                  <a
                    className="source"
                    href="http://url"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Go to source...
                  </a>,
                ],
                "className": "details",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                          Object {
                            "icon": Array [
                              536,
                              512,
                              Array [],
                              "f5c0",
                              "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                            ],
                            "iconName": "star-half-alt",
                            "prefix": "fas",
                          }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        title=""
                        transform={null}
                      />,
                      <span>
                        20
                      </span>,
                    ],
                    "className": "score",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "border": false,
                        "className": "",
                        "fixedWidth": false,
                        "flip": null,
                        "icon": Object {
                          "icon": Array [
                            536,
                            512,
                            Array [],
                            "f5c0",
                            "M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z",
                          ],
                          "iconName": "star-half-alt",
                          "prefix": "fas",
                        },
                        "inverse": false,
                        "listItem": false,
                        "mask": null,
                        "pull": null,
                        "pulse": false,
                        "rotation": null,
                        "size": null,
                        "spin": false,
                        "symbol": false,
                        "title": "",
                        "transform": null,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": 20,
                      },
                      "ref": null,
                      "rendered": 20,
                      "type": "span",
                    },
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "View metrics data...",
                    "className": "source",
                    "href": "http://url/detail",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "View metrics data...",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Go to source...",
                    "className": "source",
                    "href": "http://url",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Go to source...",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
